### ISSUE TITLE
Local Gradient Aggregation learning rate tuning

### ISSUE THREAD

**Issue by 0as1s** [horovod/horovod#2710]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet)
2. Framework version:
3. Horovod version:
4. MPI version:
5. CUDA version:
6. NCCL version:
7. Python version:
8. OS and version:
9. GCC version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before? 
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**

Hi, could anyone share some experience tuning the learning rate after adding local gradient aggregation? I tried using learning_rate * hvd.size() and just learning_rate. But they both slowed the convergency. I'm using TensorFlow and hvd 0.21.

Thanks a lot!

**Comment by tgaddair** [comment]
Hey @0as1s, typically you would want to scale your learning rate as `learning_rate * hvd.size() * batches_per_allreduce`. This is because with local gradient aggregation you are effectively increasing your batch size.

**Comment by 0as1s** [comment]
> Hey @0as1s, typically you would want to scale your learning rate as `learning_rate * hvd.size() * batches_per_allreduce`. This is because with local gradient aggregation you are effectively increasing your batch size.

Hi @tgaddair, thanks a lot for your advice! Let me try and see if this learning rate works.

According to determined's post https://determined.ai/blog/optimizing-horovod/, the training process is "doing N normal forward and backward locally and then synchronize among ranks". From my understanding, the large learning rate affects the normal local training process. Or hvd already optimizes this for us?

Please correct me if I am wrong.

**Comment by 0as1s** [comment]
> Hey @0as1s, typically you would want to scale your learning rate as `learning_rate * hvd.size() * batches_per_allreduce`. This is because with local gradient aggregation you are effectively increasing your batch size.

Hi @tgaddair, I just read the source code of local gradient aggregation. Seems the local training process just aggregates gradients instead of updating parameters. So the batch_size is multiplied by batches_per_allreduce.

Thanks again! It really helps.
